lang/cs/Org.Apache.REEF.IO/PartitionedData/FileSystem/FileSystemInputPartition.cs

```diff
@@ -54,26 +56,73 @@ internal sealed class FileSystemInputPartition<T> : IInputPartition<T>, IDisposa
             _id = id;
             _fileSystem = fileSystem;
             _fileSerializer = fileSerializer;
-            _filePaths = filePaths;
             _tempFileCreator = tempFileCreator;
-            _isInitialized = false;
-            _copyToLocal = copyToLocal;
+            _filePaths = filePaths;
+
+            if (!copyToLocal)
+            {
+                // Implies that the files are already local.
+                _localFiles = Optional<ISet<string>>.Of(filePaths);
+            }
+            else
+            {
+                _localFiles = Optional<ISet<string>>.Empty();
+            }
         }
 
         public string Id
         {
             get { return _id; }
         }
 
-        private void Initialize()
+        public void Cache()
```

Comment:
We have been talking about cache, do we? Even the method name is called Cache. That's was why I was asking in Jira this morning what does cache mean. 
There are two layers as my understanding:
1. Data file Downloading from remote to local. Currently this is configurable. 
2. Data cache, - get IEnumerable, iterate it so that it is cached.

Markus replied it is 
2. Data cache, - get IEnumerable, iterate it so that it is cached.

If this Jira is to address 1 and REEF-1357 is to address 2. Then we are fine. But may need to update the title of the Jira.